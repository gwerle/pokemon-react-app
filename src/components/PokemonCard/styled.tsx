import styled from "styled-components";

export const CardInfo = styled.div`
  display: flex;
  align-items: center;
  justify-content: space-between;
`;
