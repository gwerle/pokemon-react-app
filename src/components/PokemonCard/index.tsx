import { memo } from "react";

import usePokemon from "@/service/get/fetchPokemonUnique";
import { PokemonListI } from "@/types";

import Card from "../base/Card";
import * as S from "./styled";

type Props = {
  pokemon: PokemonListI;
};

function PokemonCard({ pokemon }: Props) {
  const { data, isLoading, error } = usePokemon(pokemon.url);
  const pokemonTypes = data?.types.map(t => t.type.name);

  if (error) {
    return <div>NÃ£o encontramos o resultado da sua pesquisa...</div>;
  }
  return (
    <li key={pokemon.url}>
      <Card imgSrc={data?.sprites.front_default || ""} isLoading={isLoading}>
        <S.CardInfo>
          <div>
            <strong>{data?.name}</strong>
          </div>
          <div>{pokemonTypes?.join("/")}</div>
        </S.CardInfo>
      </Card>
    </li>
  );
}

export default memo(PokemonCard);
