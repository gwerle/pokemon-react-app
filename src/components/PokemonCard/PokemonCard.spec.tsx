import { screen, waitFor } from "@testing-library/react";

import { renderWithTheme } from "@/utils/tests/renderWithTheme";

import PokemonCard from ".";

describe("<PokemonCard />", () => {
  it("Renders without error", () => {
    const { container } = renderWithTheme(
      <PokemonCard
        pokemon={{ name: "test", url: "https://pokeapi.co/api/v2/pokemon/1/" }}
      ></PokemonCard>
    );

    expect(container).toMatchSnapshot();
  });

  it("should show bold pokemon name", async () => {
    renderWithTheme(
      <PokemonCard
        pokemon={{ name: "test", url: "https://pokeapi.co/api/v2/pokemon/1/" }}
      ></PokemonCard>
    );

    await waitFor(() => {
      const pokemonNameEl = screen.getByText("bulbasaur");
      expect(pokemonNameEl).toBeInTheDocument();
    });
  });
});
