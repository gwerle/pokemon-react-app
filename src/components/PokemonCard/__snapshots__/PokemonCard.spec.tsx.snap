// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PokemonCard /> Renders without error 1`] = `
<div>
  <li>
    <div
      class="sc-gueYoa emUOiO"
    >
      <div
        class="sc-beqWaB iTdBVO"
        data-testid="skeleton"
        height="150px"
        width="100%"
      />
      <div
        class="sc-dmqHEX hZHgck"
      >
        <div>
          <strong />
        </div>
        <div />
      </div>
    </div>
  </li>
</div>
`;
